GoogleAnalyticsImporter.php

```diff
@@ -359,4 +367,79 @@ private function getBoundsInTimezone(\Piwik\Period $period, $timezone)
 
         return [$date1, $date2];
     }
+
+    public static function datesOverlap($periods, $start_time_key = 'start_time', $end_time_key = 'end_time')
+    {
+        // order periods by start_time
+        usort($periods, function ($a, $b) use ($start_time_key, $end_time_key) {
+            return strtotime($a[$start_time_key]) <=> strtotime($b[$end_time_key]);
+        });
+        // check two periods overlap
+        foreach ($periods as $key => $period) {
+            if ($key != 0) {
+                if (strtotime($period[$start_time_key]) < strtotime($periods[$key - 1][$end_time_key])) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+
+    /**
+     * Check if there are pending imports, and if so, if the report date is in the range of the dates of the import
+     * @return bool
+     * @throws \Exception
+     */
+    public static function canDisplayImportPendingNotice(): bool
+    {
+        if(!self::hasOverLapCheckParams()){
+            return false;
+        }
+
+        $currentIdSite = Common::getRequestVar('idSite', -1);
+        if($currentIdSite === -1){
+            return false;
+        }
+
+        $instance = new ImportStatus();
+        try{
+            $status = $instance->getImportStatus($currentIdSite);
+            if ($status['status'] != ImportStatus::STATUS_ONGOING) {
+                return false;
+            }
+        } catch (\Exception $exception){
+            return false;
+        }
+
+
+        $periods = [
+            //Report Dates
+            [
+                'start_time' => Date::factory(Period\Factory::makePeriodFromQueryParams('',
+                    Common::getRequestVar('period'), Common::getRequestVar('date'))->getDateStart()),
+                'end_time' => Date::factory(Period\Factory::makePeriodFromQueryParams('',
+                    Common::getRequestVar('period'), Common::getRequestVar('date'))->getDateEnd())
+            ],
+            //Import Dates
+            [
+                'start_time' =>Date::factory($status['import_range_start']),
+                'end_time' => Date::factory($status['import_range_end'])
+            ]
+        ];
+
+        if(self::datesOverlap($periods)) {
+            return true;
+        }
+        return false;
+    }
+
+    public static function hasOverLapCheckParams()
```

Comment:
@elabuwa Didn't understood why we need this check ?

Reply:
The notification will be shown only if the below conditions meet @AltamashShaikh 
1. An importer task exists and not finished
2. The import tasks date range overlaps the report date range
Can you double check the importer date range overlaps your view/report/filter's date range?
Eg: If the importer is importing between 2022-11-01 and 2022-11-05 and your filtered date is between 2022-11-06 and 2022-11-10 it won't show the notice. However, if your filtered date is between 2022-11-04 and 2022-11-10 it will show the notice because there is overlap in the 2 date ranges.

Reply:
It works now :+1: